src/buildercore/cfngen.py

```diff
@@ -263,9 +286,21 @@ def build_context_fastly(context, parameterize):
     def _build_subdomain(x):
         return complete_domain(parameterize(x), context['domain'])
 
-    def _parameterize_hostname(b):
-        b['hostname'] = parameterize(b['hostname'])
-        return b
+    def _build_shield(shield):
+        if shield is False:
+            return {}
+
+        if shield is True:
+            pop = FASTLY_AWS_REGION_SHIELDS.get(context['project']['aws']['region'], 'us-east-1')
+
+            return {'pop': pop}
+
+        return shield
+
+    def _build_backend(backend):
+        backend['hostname'] = parameterize(backend['hostname'])
+        backend['shield'] = _build_shield(backend.get('shield', context['project']['aws']['fastly'].get('shield', False)))
```

Comment:
this seems to be information that is independent of the system provisioning it (Terraform, API calls, whatever) so good fit for the `context`